### PR TITLE
Chore/weather forecast city

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description
This PR contains changes from https://github.com/superfaceai/station/pull/69. It uses superface/testing for tests.
There is a weird Validation issue which needs to be resolved.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Jakub-Vacek** [comment]
Validator is expecting Enum to be ListStructure. I think this is scoped only to operations called with foreach - (`current-city` capability map for the same provider uses same enum without the foreach call and it passes validation).

@freaz @janhalama did you see this issue in other places / do you have any ideas how to work around it? 

Snímek obrazovky 2021-11-08 v 9 28 43

File: capabilities/weather/forecast-city/maps/forecast-city.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        it('should return weather object', async () => {
+          const input = {
+            city: 'Prague,CZ',
+          };
+
+          await expect(
+            superface.run(
+              {
+                profile: 'weather/forecast-city',
+                provider,
```
Comment (martinalbert) [inline capabilities/weather/forecast-city/maps/forecast-city.ts:30]: This is repeating, you can pass them once in constructor of `SuperfaceTest`.

**Review by martinalbert — APPROVED** [review]

**Review by martinalbert — APPROVED** [review]

**Review by martinalbert — APPROVED** [review]

**Comment by Jakub-Vacek** [comment]
@janhalama @freaz 🙏 I think this is good to go :)

**Comment by janhalama** [comment]
> Validator is expecting Enum to be ListStructure. I think this is scoped only to operations called with foreach - (`current-city` capability map for the same provider uses same enum without the foreach call and it passes validation).
> 
> @freaz @janhalama did you see this issue in other places / do you have any ideas how to work around it?
> 
> Snímek obrazovky 2021-11-08 v 9 28 43

@Jakub-Vacek looks like the variable has to be in plural -> `units`

File: capabilities/weather/forecast-city/profile.supr [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  ]
+  error {
+    """
+    Message
+    A short, human-readable summary of the problem.
+    """
+    message string!
+    """
+    Status code
+    Usually a HTTP status code connected to occured problem.
```
Comment (janhalama) [inline capabilities/weather/forecast-city/profile.supr:63]: Profile should not contain any details about the API. I would avoid returning HTTP status code here.

**Review by janhalama — COMMENTED** [review]
I am not sure about the error response. Otherwise looks good.

**Comment by janhalama** [comment]
> Validator is expecting Enum to be ListStructure...
> Did you see this issue in other places / do you have any ideas how to work around it?

@Jakub-Vacek I saw similar issue. The order of operation arguments solved it. But it doesn't help here 🤷‍♂️

File: capabilities/weather/forecast-city/profile.supr [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      maxTemperature number!
+
+      """
+      MinTemperature
+      Datily minimal temperature in specified units
+      """
+      minTemperature number!
+    }
+  ]
+  error {
```
Comment (freaz) [inline capabilities/weather/forecast-city/profile.supr:55]:
```suggestion

  error {
```

File: capabilities/weather/forecast-city/profile.supr [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    """
+    Message
+    A short, human-readable summary of the problem.
+    """
+    message string!
```
Comment (freaz) [inline capabilities/weather/forecast-city/profile.supr:60]: Shouldn't be also message itself required?

```suggestion
    message! string!
```

**Review by freaz — APPROVED** [review]
Looks fine, two minor formatting things and not sure about error model.

File: capabilities/weather/forecast-city/profile.supr (hunk 4 above)
Comment (Jakub-Vacek) [reply capabilities/weather/forecast-city/profile.supr]: I changed error to title and detail object we use is other profiles.

**Pull request merged**